integration/test/migration/migration_test.go

```diff
@@ -0,0 +1,367 @@
+// before installing managed version of the metrics-server.
+func TestMigration(t *testing.T) {
+	ctx := context.Background()
+
+	// Install legacy resources.
+	err := helmClient.InstallFromTarball("/e2e/fixtures/resources-chart", resourceNamespace, helm.ReleaseName("resources"))
+	if err != nil {
+		t.Fatalf("could not install resources chart: %v", err)
+	}
+
+	// Check legacy resources are present.
+	err = checkLegacyResourcesPresent()
+	if err != nil {
+		t.Fatalf("legacy resources not present: %v", err)
+	}
+
+	// Check managed resources are not present.
+	err = checkManagedResourcesNotPresent("app=metrics-server,giantswarm.io/service-type=managed")
+	if err != nil {
+		t.Fatalf("managed resources present: %v", err)
+	}
+
+	// Install kubernetes-metrics-server-chart.
+	channel := fmt.Sprintf("%s-%s", env.CircleSHA(), testName)
+	err = r.Install(chartName, templates.MetricsServerValues, channel)
+	if err != nil {
+		t.Fatalf("could not install %q %v", chartName, err)
+	}
+
+	// Wait for deployed status
+	err = r.WaitForStatus(chartName, "DEPLOYED")
+	if err != nil {
+		t.Fatalf("timeout waiting for deployed resource: %v", err)
+	}
+
+	// Check legacy resources are not present.
+	err = checkLegacyResourcesNotPresent("app!=metrics-server,giantswarm.io/service-type!=managed")
+	if err != nil {
+		t.Fatalf("legacy resources present: %v", err)
+	}
+
+	// Check managed resources are present.
+	err = checkManagedResourcesPresent("app=metrics-server,giantswarm.io/service-type=managed")
+	if err != nil {
+		t.Fatalf("managed resources not present: %v", err)
+	}
+
+	// Check metrics availability
+	err = checkMetricsAvailability(ctx)
+	if err != nil {
+		t.Fatalf("could not get metrics: %v", err)
+	}
+
+	// Delete release
+	err = helmClient.DeleteRelease(chartName, helm.DeletePurge(true))
+	if err != nil {
+		t.Fatalf("failed to teardown resource: %v", err)
+	}
+}
+
+func checkMetricsAvailability(ctx context.Context) error {
+	var err error
+
+	restClient := h.K8sClient().CoreV1().RESTClient()
+
+	l.LogCtx(ctx, "level", "debug", "message", "waiting for the metrics become available")
+
+	o := func() error {
+
+		_, err := restClient.Get().RequestURI(metricsAPIEndpoint).Stream()
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+	b := backoff.NewConstant(longMaxInterval, shortMaxInterval)
+	n := backoff.NewNotifier(l, ctx)
+
+	err = backoff.RetryNotify(o, b, n)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	l.LogCtx(ctx, "level", "debug", "message", "successfully retrieved metrics from metrics server")
+
+	return nil
+}
+
+func checkLegacyResourcesPresent() error {
+	var err error
+
+	c := h.K8sClient()
+	ac := h.K8sAggregationClient()
+	getOptions := metav1.GetOptions{}
+
+	_, err = c.Rbac().ClusterRoleBindings().Get("metrics-server:system:auth-delegator", getOptions)
+	if err != nil {
+		return microerror.Newf("failed to get clusterrolebinding %s: %v", "metrics-server:system:auth-delegator", err)
```

Comment:
Not a blocker but current state of the art in tests is to use `%#q` for names. Output is quoted so its easier to read. The #q makes it more reliable but I forget why!